iceoryx_hoofs/reporting/include/iox/error_reporting/errors.hpp

```diff
@@ -36,9 +36,21 @@ static constexpr const char* UNKNOWN_ERROR_NAME = "unknown error";
 // By default, there are only error codes and violations.
 // Custom errors can be added but must satisfy the minimal interface.
 
+enum class ViolationErrorCode : uint32_t
+{
+    REQUIRED_CONDITION_VIOLATION,
+    ASSUMPTION_VIOLATION,
+    PRECONDITION_VIOLATION
+};
+
 class Violation
```

Comment:
@elBoberido I was also considering having distinct types for everything, but decided against it as they are essentially all wrapper around a code for now. It should be considered if this changes.